packages/transaction_signing_gateway/lib/storage/cosmos_key_info_storage.dart

```diff
@@ -233,4 +233,12 @@ class CosmosKeyInfoStorage implements KeyInfoStorage {
       );
     }
   }
+
+  @override
+  Future<Either<CredentialsStorageFailure, bool>> clearCredentials() async {
+    final _secureCleared = await _secureDataStore.clearAllData();
+    final _plainCleared = await _plainDataStore.clearAllData();
+    _secureCleared.fold(left, right);
+    return _plainCleared.fold(left, right);
```

Comment:
Secure and plain data clearing should be in separate methods both returning `Future<Either<CredentialsStorageFailure, bool>>` and both of them separately being called in this `clearCredentials` function.

Reply:
Also, this should be wrapped in try catch block.